### DISCUSSION TITLE
Country label to span country's countour

### DISCUSSION THREAD

**Discussion by Shevick** [maplibre/maplibre-gl-js#1834] [Q&A]
Hello all,

The default country labels are very boring. I would like to make them larger to fill up the country's borders. I'd like it to look more like this: [img]https://i.ibb.co/WKCCjXJ/Label-Span.png[/img]

Looking at my options in maplibre style editor, it looks like there is an option for making the label placement follow a line or 'line-center'. When I try these options, my label disappears. I think this is due to my source not using linestring or polygon geometries, which the tool tip says are required to use those options over the default, 'point'. I'm using openmaptiles as my source. 

I'm thinking if I convert the coordinates for every country's polygon to a line (find the mid-point for every coordinate pair?) I could assign the label to that line's geometry. I could also possibly place it as a symbol so it sticks to the map while panning?

Is that viable? My other option I was thinking is to use illustrator to export an image of the text and use it as a symbol/marker at the label's coordinates. 

Both bad ideas? Other thoughts?

**Comment by HarelM** [comment]
Since countries do not change a lot I would simply draw a line for every county and keep that as a source for the line labels as a geojson source. I would avoid images as those will be a lot harder to get right for different zoom levels.
